### PR TITLE
Fix #298 Handle removal of sklearn randomized l1 models

### LINKED ISSUE #298: Randomized L1 models are removed from sklearn, breaks import
The Randomized Linear models were removed from sklearn but it looks like they are still being imported. That breaks the import.

### PR DESCRIPTION
**Pull request by teabolt**
Fixes #298 
Wrap import and decorators in a try-except block.

### CODE REVIEW

File: eli5/sklearn/transform.py [hunk 1]
```diff
@@ -21,15 +18,23 @@
...
+try:
+    from sklearn.linear_model import (  # type: ignore
+        RandomizedLogisticRegression,
+        RandomizedLasso,
+    )
+    _select_names = transform_feature_names.register(RandomizedLasso)(_select_names)
+    _select_names = transform_feature_names.register(RandomizedLogisticRegression)(_select_names)
+except ImportError:     # Removed in scikit-learn 0.21
+    pass
+_select_names = transform_feature_names.register(SelectorMixin)(_select_names)
```
Comment (lopuhin) [inline eli5/sklearn/transform.py]: ``SelectorMixin`` could be left as a decorator, right?

File: eli5/sklearn/transform.py [hunk 2]
```diff
@@ -21,15 +18,23 @@
 
 # Feature selection:
 
-@transform_feature_names.register(SelectorMixin)
-@transform_feature_names.register(RandomizedLogisticRegression)
-@transform_feature_names.register(RandomizedLasso)
```
Comment (lopuhin) [inline eli5/sklearn/transform.py]: I had an idea for another approach here, but after some thought it looks worse than what is implemented here, still let me put it here:
- put imports at the top, set ``RandomizedLasso `` etc. to ``None`` if they fail to import
- make it possible to write something equivalent to ``@transform_feature_names.register(RandomizedLasso)`` which would do nothing if ``RandomizedLasso`` is ``None``. But it seems that we'd need a wrapper for that, something like ``@register(transform_feature_names, RandomizedLasso)``

Advantage is that the flow would be more natural (imports at the top, decorators above the function). Disadvantage is that it might be a bit confusing that ``RandomizedLasso`` can be ``None``, and also we would need to introduce the wrapper. So to me it seems that it would make sense only if we get a lot of similar conditional registers.

File: eli5/sklearn/transform.py (hunk 1 above)
Comment (teabolt) [reply eli5/sklearn/transform.py]: You are correct. I expanded out and placed `SelectorMixin` last to keep the original order that the decorators are applied in (RandomizedLasso -> RandomizedLogisticRegression -> SelectorMixin). But in this case order does not seem to matter.

File: eli5/sklearn/transform.py (hunk 2 above)
Comment (teabolt) [reply eli5/sklearn/transform.py]: Thanks for the suggestions @lopuhin. Looking at the [recently deprecated](https://scikit-learn.org/stable/modules/classes.html#recently-deprecated) in sklearn there doesn't seem to be too many removals of existing models. But I would certainly prefer your approach if we face more issues like this down the line.
